packages/shared/src/components/sources/SourceCard.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { ReactElement, ReactNode } from 'react';
+import classNames from 'classnames';
+import { useRouter } from 'next/router';
+import SquadMemberShortList from '../squads/SquadMemberShortList';
+import { Card } from '../cards/Card';
+import { Button } from '../buttons/Button';
+import { Connection } from '../../graphql/common';
+import {
+  SourceMember,
+  SourceMemberRole,
+  SourceType,
+} from '../../graphql/sources';
+import { Image } from '../image/Image';
+import { SquadDirectoryCardBannerDefault } from '../../lib/constants';
+import { SquadJoinButton } from '../squads/SquadJoinButton';
+import { Origin } from '../../lib/analytics';
+
+interface SourceCardProps {
+  title: string;
+  subtitle?: string;
+  icon?: ReactNode;
+  image?: string;
+  description?: string;
+  action?: {
+    text: string;
+    onClick?: () => void;
+  };
+  link?: {
+    text: string;
+    href: string;
+    target?: string;
+  };
+  members?: Connection<SourceMember>;
+  membersCount?: number;
+  permalink?: string;
+  id?: string;
+  banner?: string;
+  borderColor?: string;
+  type?: SourceType;
+  handle?: string;
+  memberInviteRole?: SourceMemberRole;
+  memberPostingRole?: SourceMemberRole;
+}
+
+export const SourceCard = ({
+  title,
+  subtitle,
+  icon,
+  image,
+  description,
+  action,
+  link,
+  members,
+  membersCount,
+  permalink,
+  id,
+  banner,
+  borderColor,
+  type,
+  handle,
+  memberInviteRole,
+  memberPostingRole,
+}: SourceCardProps): ReactElement => {
+  const router = useRouter();
+
+  const squad = {
+    name: title,
+    description,
+    members,
+    membersCount,
+    permalink,
+    id,
+    active: true,
+    public: true,
+    image,
+    handle,
+    memberInviteRole,
+    memberPostingRole,
+  };
+
+  return (
+    <Card
+      className={classNames(
+        'overflow-hidden p-0',
+        borderColor
+          ? `border-theme-color-${borderColor} hover:border-theme-color-${borderColor}`
+          : 'border-theme-color-avocado hover:border-theme-color-avocado',
+      )}
+    >
+      <div className="h-24 rounded-t-2xl bg-theme-bg-onion">
+        <Image
+          className="object-cover w-full h-full"
+          src={banner || SquadDirectoryCardBannerDefault}
+          alt="Banner image for source, showing abstract colors."
+        />
+      </div>
+      <div className="flex flex-col flex-1 p-4 -mt-12 rounded-t-2xl bg-theme-bg-secondary">
+        <div className="flex justify-between items-end mb-3">
+          {!!image && (
+            <img
+              className="-mt-14 w-24 h-24 rounded-full z-10"
+              src={image}
+              alt={`${title} source`}
+            />
+          )}
+          {!image && (
+            <div className="flex justify-center items-center -mt-14 w-24 h-24 bg-pepper-40 rounded-full z-10">
```

Comment:
This is the icon one right, I think we said to use the pepper.40 value named to switch reverse in light mode right?